src/core/ext/transport/chttp2/server/insecure/server_chttp2_posix.c

```diff
@@ -57,7 +60,8 @@ void grpc_server_add_insecure_channel_from_fd(grpc_server *server, int fd) {
   const grpc_channel_args *server_args = grpc_server_get_channel_args(server);
   grpc_transport *transport = grpc_create_chttp2_transport(
       &exec_ctx, server_args, server_endpoint, 0 /* is_client */);
-  grpc_server_setup_transport(&exec_ctx, server, transport, server_args);
+  grpc_endpoint_add_to_pollset(&exec_ctx, server_endpoint, grpc_cq_pollset(cq));
```

Comment:
I think you can drop the extra parameter, and just pass through NULL here for the pollset without any problem.